### ISSUE TITLE
Cache Directory Being Written One Directory Higher than Where Minetest is Executed From

### ISSUE THREAD

**Issue by LazyJ** [minetest/minetest#3463]
**Minetest Engine (2015_12_17):**  bde6194
**Minetest Game (2015_12_12):** [b57ecb9](https://github.com/minetest/minetest_game/commit/b57ecb94df2bfc1c6383a3d35465e22b683821ee)
**OS:** Linux Mint 17.3

For the past couple years, I have been using a script to start Minetest from anywhere in the terminal.
```
#!/bin/bash

~/minetest_stuff/minetest/bin/./minetest --logfile ''
```
Recently, I noticed "cache" beginning to appear in odd places in my computer. After some testing, I found that Minetest was now writing the cache one directory above from where ever Minetest was started from. 

Test:
1. File path:    ~/minetest_stuff/test1/test2/test3/minetest/bin/
2. cd to test1
3. Execute ./minetest from test1:  test2/test3/minetest/bin/./minetest

Result:
* No cache directory created in ~/minetest_stuff/test1/test2/test3/minetest/
* cache is created in ~/minetest_stuff/

**Comment by est31** [comment]
@LazyJ do you do an in-place build?

@ShadowNinja this seems to be a regression of ea2964f5a168cb52d1b9f74a08f00c7c068c6649

**Comment by rubenwardy** [comment]
side note: the ./ is pointless (it means "go to the current dir" not "run executable") afaik: `~/minetest_stuff/minetest/bin/minetest`.

**Comment by ShadowNinja** [comment]
I think I found the issue: if a `XDG_CACHE_HOME` can't be found or `RUN_IN_PLACE` is used, `path_cache` is left at its default of `$PATH_USER/cache` (where PATH_USER is `..`) instead of being reset to `$PATH_USER/cache` after `PATH_USER` is set properly.

**Comment by LazyJ** [comment]
@est31, yes it is an in-place build. I run four, separate builds of Minetest for various purposes. Each is compiled from the same release as the other three, are compiled as in-place, each has its own start-up script, and each is now writing "cache" one directory above where the scripts were executed from.

@rubenwardy, I've gotten into the habit of using "./"  to preface a script when in the script's directory or, in as in this case, using it in the file-path line of script to run another script or executable *(? -I'm  not sure on the terminology)* file in Linux because it "just works" where as in the past that has not always been the case for me.

**Comment by est31** [comment]
Fixed by c6bb6f99d1fc0acd0161307a17dd31eea4a56fff

**Comment by est31** [comment]
Ah sorry, mentioned wrong commit. Its 9719aded5430dd57591310c9cae06944dc345d42 of course.
